app/components/click_observer_component.rb

```diff
@@ -1,14 +1,22 @@
 # frozen_string_literal: true
 
 class ClickObserverComponent < BaseComponent
-  attr_reader :event_name, :tag_options
+  attr_reader :event_name, :tag_options, :payload
 
-  def initialize(event_name:, **tag_options)
+  def initialize(event_name:, payload: {}, **tag_options)
```

Comment:
I think we could default `nil` here and it would work just as well, and would better represent a "nothing" value and avoid unnecessary object allocations and JSON serialization.

```suggestion
  def initialize(event_name:, payload: nil, **tag_options)
```